packages/samples/specs/resource-manager/operations/operation-provider/main.tsp

```diff
@@ -0,0 +1,70 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.ResourceManager;
+using Azure.Core;
+
+@service({
+  title: "Microsoft.OperationsTest",
+})
+@versioned(Versions)
+@armProviderNamespace
+namespace Microsoft.OperationStatusSample;
+
+enum Versions {
+  @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @armCommonTypesVersion(Azure.ResourceManager.CommonTypes.Versions.v5)
+  `2022-11-01-preview`,
+}
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+/** Represents a virtual machine size */
+model VmSize {
+  /** The name of the virtual machine size */
+  name: string;
+}
+
+/** LogAnalytics operation status response */
+model LogAnalyticsOperationResult {
+  /** LogAnalyticsOutput */
+  output: string;
+}
+
+@armResourceOperations
+interface ProviderOperations {
+  /** Operation to get virtual machines for subscription (/subscriptions/{subscriptionId}/providers/Microsoft.ContosoProviderHub/getVmSizes) */
+  @get
+  getVmSizes is ArmProviderActionAsync<Response = Page<VmSize>, Scope = SubscriptionActionScope>;
```

Comment:
```suggestion
  getVmSizes is ArmProviderActionAsync<Response = VmSizeCollection>;
```

You should create a separate model for each of the collection types
```
model VmSizeCollection = Page<VmSize>;
```

This allows clients to control the name of the collection type.  You should do this for each of the operations below.  In these samples, we need to model best practices for ARM libraries.